src/main/java/org/mule/extensions/jms/internal/source/JmsListener.java

```diff
@@ -210,26 +210,30 @@ public void onStart(SourceCallback<Object, JmsAttributes> sourceCallback) throws
         }
 
         JmsListenerLock jmsLock = createJmsLock();
-        sessions.add(new Pair<>(session, jmsLock));
+        createdListeners.add(new MessageListenerInfo(session, jmsLock, consumer));
         consumer.listen(messageListenerFactory.createMessageListener(session, jmsLock));
       }
     } catch (Exception e) {
-      String msg = format("An error occurred while creating the consumers for destination [%s]: %s",
-                          destination, e.getMessage());
+      String msg = format("An error occurred while creating the consumers for destination [%s:%s]: %s",
+                          getDestinationType(), destination, e.getMessage());
       LOGGER.error(msg, e);
+      releaseListeners();
+
+      if (e.getCause() instanceof ConnectException) {
```

Comment:
Use ExceptionUtils.extractConnectionException() instead

Reply:
this is a `java.net.ConnectException`, not a `ConnectionException`